src/gui/processing/qgsprocessingmaplayercombobox.cpp

```diff
@@ -613,6 +650,14 @@ void QgsProcessingMapLayerComboBox::onLayerChanged( QgsMapLayer *layer )
     }
   }
 
+  if ( mParameter->type() == QgsProcessingParameterRasterLayer::typeName() )
+  {
+    // Only WMS layers whose parameter supports WmsScale and WmsDpi will access raster advanced options for now.
+    QgsProcessingParameterRasterLayer *rasterParameter = static_cast<QgsProcessingParameterRasterLayer *>( mParameter.get() );
+    const bool supportsRasterOptions = rasterParameter->parameterCapabilities() & Qgis::RasterProcessingParameterCapability::WmsScale && rasterParameter->parameterCapabilities() & Qgis::RasterProcessingParameterCapability::WmsDpi;
```

Comment:
This should be an "or":

```suggestion
    const bool supportsRasterOptions = rasterParameter->parameterCapabilities().testFlag( Qgis::RasterProcessingParameterCapability::WmsScale ) || rasterParameter->parameterCapabilities().testFlag( Qgis::RasterProcessingParameterCapability::WmsDpi );
```